llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -190,29 +190,35 @@ void PostPipeliner::computeLoopCarriedParameters() {
   }
 
   // Propagate Earliest upstream, initialize Latest
+  // Unrestricted: last cycle of last stage
+  const int Latest = NCopies * II - 1;
```

Comment:
This has been changed. We now have negative values representing the (reverse) distance to the end. It matters since we also use it to define the range where to insert.